Neos.ContentRepository.Core/Classes/Factory/ContentRepositoryFactory.php

```diff
@@ -121,7 +122,7 @@ public function buildService(
     ): ContentRepositoryServiceInterface {
         $serviceFactoryDependencies = ContentRepositoryServiceFactoryDependencies::create(
             $this->projectionFactoryDependencies,
-            $this->build(),
+            $this->getOrBuild(),
```

Comment:
like `getOrBuildService`

Reply:
We thought about it, but there was some issue with that (@skurfuerst do you remember what it was?)

That's why we wrote

> As a result that would mean that CR Services are not singletons but built whenever ContentRepositoryRegistry::buildService() is called

above (https://github.com/neos/neos-development-collection/pull/4310#issuecomment-1611171245)

But, good point.. maybe we should re-consider

Reply:
@bwaidelich they cannot be singletons because you need to be able to create the same service twice with different parameters (passed from the service factory)